pkg/reconciler/broker/broker_test.go

```diff
@@ -814,3 +1163,125 @@ func createTriggerChannelRef() *corev1.ObjectReference {
+}
+func makeServiceURI() *apis.URL {
+	return &apis.URL{
+		Scheme: "http",
+		Host:   fmt.Sprintf("%s.%s.svc.%s", makeBrokerFilterService().Name, testNS, utils.GetClusterDomainName()),
+		Path:   fmt.Sprintf("/triggers/%s/%s/%s", testNS, triggerName, triggerUID),
+	}
+}
+func makeEmptyDelivery() *eventingduckv1alpha1.DeliverySpec {
+	return nil
+}
+func makeBrokerFilterService() *corev1.Service {
+	return resources.MakeFilterService(makeBroker())
+}
+
+func makeBroker() *v1alpha1.Broker {
+	return &v1alpha1.Broker{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "eventing.knative.dev/v1alpha1",
+			Kind:       "Broker",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: testNS,
+			Name:      brokerName,
+		},
+		Spec: v1alpha1.BrokerSpec{},
+	}
+}
+
+func allBrokerObjectsReadyPlus(objs ...runtime.Object) []runtime.Object {
+	brokerObjs := []runtime.Object{
+		NewBroker(brokerName, testNS,
+			WithBrokerChannel(channel()),
+			WithInitBrokerConditions,
+			WithBrokerReady,
+			WithBrokerTriggerChannel(createTriggerChannelRef()),
+			WithBrokerAddress(fmt.Sprintf("%s.%s.svc.%s", ingressServiceName, testNS, utils.GetClusterDomainName()))),
+		createChannel(testNS, triggerChannel, true),
+		NewDeployment(filterDeploymentName, testNS,
+			WithDeploymentOwnerReferences(ownerReferences()),
+			WithDeploymentLabels(resources.FilterLabels(brokerName)),
+			WithDeploymentServiceAccount(filterSA),
+			WithDeploymentContainer(filterContainerName, filterImage, livenessProbe(), readinessProbe(), envVars(filterContainerName), containerPorts(8080))),
+		NewService(filterServiceName, testNS,
+			WithServiceOwnerReferences(ownerReferences()),
+			WithServiceLabels(resources.FilterLabels(brokerName)),
+			WithServicePorts(servicePorts(8080))),
+		NewDeployment(ingressDeploymentName, testNS,
+			WithDeploymentOwnerReferences(ownerReferences()),
+			WithDeploymentLabels(resources.IngressLabels(brokerName)),
+			WithDeploymentServiceAccount(ingressSA),
+			WithDeploymentContainer(ingressContainerName, ingressImage, livenessProbe(), nil, envVars(ingressContainerName), containerPorts(8080))),
+		NewService(ingressServiceName, testNS,
+			WithServiceOwnerReferences(ownerReferences()),
+			WithServiceLabels(resources.IngressLabels(brokerName)),
+			WithServicePorts(servicePorts(8080))),
+	}
+	return append(brokerObjs[:], objs...)
+}
+
+// Just so we can test subscription updates
+func makeDifferentReadySubscription() *messagingv1alpha1.Subscription {
+	s := makeIngressSubscription()
+	s.Spec.Subscriber.URI = apis.HTTP("different.example.com")
+	s.Status = *v1alpha1.TestHelper.ReadySubscriptionStatus()
+	return s
+}
+
+func makeIngressSubscriptionNotOwnedByTrigger() *messagingv1alpha1.Subscription {
+	sub := makeIngressSubscription()
+	sub.OwnerReferences = []metav1.OwnerReference{}
+	return sub
+}
+
+func makeReadySubscription() *messagingv1alpha1.Subscription {
+	s := makeIngressSubscription()
+	s.Status = *v1alpha1.TestHelper.ReadySubscriptionStatus()
+	return s
+}
```

Comment:
perhaps reuse some of the `trigger_test.go` bits ?
https://github.com/knative/eventing/blob/master/pkg/reconciler/trigger/trigger_test.go#L1170-L1174

Reply:
Yep, done :)